src/persist/src/indexed/mod.rs

```diff
@@ -198,13 +199,29 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
                 entry
                     .updates
                     .iter()
-                    .map(|((key, val), ts, diff)| (seqno, (key.clone(), val.clone()), *ts, *diff)),
+                    .map(|((key, val), ts, diff)| ((key.clone(), val.clone()), *ts, *diff)),
             );
+
+            // Sanity check that each sequence number occurs exactly once.
+            let prev = sequence_numbers.insert(seqno);
+            debug_assert_eq!(prev, false);
             Ok(())
         })?;
+
+        if cfg!(any(debug, test)) {
```

Comment:
my only idea for how to test this rn is to either have a faulty buffer implementation or ... I dunno everything else I can think of would require a more intrusive breaking apart of the api. Suggestions welcome here

Reply:
break the method in half and call the second part directly from the test?

Reply:
I ended up just extracting this sanity check into its own function and testing it separately. also, I decided to enable the sanity check for all builds because imo iterating over a hashset and doing a series of range checks is extremely low overhead compared to the rest of this function which does a lot more sorting and copying for each corresponding sequence number in the set of sequence numbers